vllm_spyre/platform.py

```diff
@@ -520,3 +527,215 @@ def get_max_output_tokens(self, prompt_len: int) -> int:
                 max_new_tokens = max(max_new_tokens, shape['new_tokens'])
 
         return max_new_tokens
+
+    @classmethod
+    def _handle_disable_compilation(cls, vllm_config: VllmConfig,
+                                    is_decoder: bool):
+        """
+        For decoder models we want to respect the `DISABLE_COMPILATION`
+        environment variable, which disallows torch_sendnn from compiling new
+        graphs and only allows it to load pre-compiled graphs.
+        In order to do this, we must load up some config from the torch_sendnn
+        cache and check to make sure that the current vllm config matches,
+        otherwise the cached artifacts cannot be used.
+
+        For encoder models, we do not allow disabling compilation.
+        """
+        disable_compilation_env_var = "DISABLE_COMPILATION"
+
+        disable_flag = os.getenv(disable_compilation_env_var, "").lower()
+        if disable_flag not in ("1", "true"):
+            return
+
+        # If this isn't a decoder model, re-enable compilation
+        if not is_decoder:
+            logger.info("Unsetting %s because %s is not a decoder model",
+                        disable_compilation_env_var,
+                        vllm_config.model_config.model)
+            os.environ[disable_compilation_env_var] = "0"
+            return
+
+        # If the user asked to disable compilation, then we need to enforce that
+        # they setup their cache
+        torch_cache_dir = os.getenv("TORCH_SENDNN_CACHE_DIR", None)
+        torch_cache_enabled = bool(
+            int(os.getenv("TORCH_SENDNN_CACHE_ENABLE", "0")))
+
+        if not torch_cache_dir or not torch_cache_enabled:
+            raise ValueError(
+                "DISABLE_COMPILATION=1 requires setting TORCH_SENDNN_CACHE_DIR "
+                "to a valid path and setting TORCH_SENDNN_CACHE_ENABLE=1")
+
+        compilation_config_path = Path(
+            torch_cache_dir) / PRE_COMPILE_MODEL_CONFIG_FILENAME
+        compilation_catalog_path = Path(
+            torch_cache_dir) / PRE_COMPILE_MODEL_CATALOG_FILENAME
+
+        if not compilation_catalog_path.exists() and \
+            not compilation_config_path.exists():
+            raise ValueError(
+                f"DISABLE_COMPILATION=1 was set, but no pre-compiled model "
+                f"config was found in the TORCH_SENDNN_CACHE_DIR: "
+                f"{str(compilation_config_path)} or"
+                f"{str(compilation_catalog_path)} does not exist")
+
+        if not compilation_catalog_path.is_file() and \
+            not compilation_config_path.is_file():
+            raise ValueError(
+                "DISABLE_COMPILATION=1 was set, but the pre-compiled model "
+                "config is not a file")
```

Comment:
These errors forces use of pre-compiled model configs, which I don't think we want (eg. so users can compile their own models by just running vllm).

Reply:
My intention was, since they pointed to a pre-compiled model and we have something there, but its not a file (may be some copying issue), then this would allow them to catch

Reply:
IMO, "since they pointed to a pre-compiled model" is not true. The configurations to the user are "disable compilation" and "use a cache", neither of which necessitate using a precompiled artifact config or catalog.

I suppose we could update the checks to say "if it exists, it must be a file" to catch the case that it ends up as a directory or something.

Reply:
I think you mean the change of the env var to `REQUIRE_PRECOMPILED_DECODERS`?
But yeah, that solves my concern by making the request to use precompiled artifacts explicit